src/families/solana/api/traced.ts

```diff
@@ -0,0 +1,99 @@
+import { log } from "@ledgerhq/logs";
+import { getEnv } from "../../../env";
+import { ChainAPI } from "./chain";
+
+export function traced(api: ChainAPI): ChainAPI {
+  const state = {
+    reqId: 0,
+    reqStartTime: new Map<number, number>(),
+  };
+
+  const startReqTrace = () => {
+    state.reqId += 1;
+    state.reqStartTime.set(state.reqId, Date.now());
+    return state.reqId;
+  };
+
+  const stopReqTrace = (reqId: number) => {
+    const reqStartTime = state.reqStartTime.get(reqId);
+    if (reqStartTime === undefined) {
+      log("warn", `request start time not found for request id <${reqId}>`);
+      return {
+        duration: 0,
+      };
+    }
+    state.reqStartTime.delete(reqId);
+    return {
+      duration: Date.now() - reqStartTime,
+    };
+  };
+
+  const proxy: ChainAPI = new Proxy(api, {
+    get(target, propKey, receiver) {
+      if (typeof propKey === "symbol") {
+        throw new Error("symbols not supported");
+      }
+      const targetValue = Reflect.get(target, propKey, receiver);
+      if (typeof targetValue === "function") {
+        return function (...args: unknown[]) {
+          const reqId = startReqTrace();
+          log("network", formatMsg({ reqId, msg: `calling <${propKey}>` }), {
+            args,
+          });
```

Comment:
One useful thing we want to log is the URL of each network call, ideally with path params as well if possible.
I don't see that explicitely logged, is it part of the `args` here?

Reply:
Yep. The problem here is at this level we can't see the URL, only function name and args. The args includes what will be sent via http. At the same time there is another logger used to log actual http calls (where we have access to the actual URL) - `httpRequestLogger` in `js.ts`

Reply:
The reason why there are two places when things are logged is because at the place where we have access to the actual URL we can't see the result of the operation. And at the place where we can see the result of the operation - we don't have access to the URL. Well, yes, it would be very nice to just provide your configured axios as a fetch client to Solana SDK, but, unfortunately, Solana SDK doesn't accept a custom fetch client yet.

Reply:
Thanks for the explanation and sorry for the delay here.
So can you please just describe the scope of this PR, what additional information will or will not be logged with it? Maybe give a sample log to illustrate.
Thanks!